rust/lance/src/dataset/fragment.rs

```diff
@@ -1380,11 +1333,14 @@ impl FragmentReader {
         params: impl Into<ReadBatchParams> + Clone,
         projection: &Schema,
     ) -> Result<RecordBatch> {
-        let read_tasks = self
-            .readers
-            .iter()
-            .enumerate()
-            .map(|(reader_idx, (reader, schema))| {
+        let first_reader = self.readers[0].0.as_legacy();
+        let batch_offset = (0..batch_id)
+            .map(|i| first_reader.num_rows_in_batch(i as i32))
+            .sum();
```

Comment:
Hmm...can you rely on a constant batch size here to do this as multiplication?

Reply:
Oh right. I could do that.